driver-sync/src/main/com/mongodb/client/internal/ClientSessionImpl.java

```diff
@@ -123,18 +120,13 @@ public void close() {
         }
     }
 
-
+    private boolean canCommitOrAbort() {
+        return !inTransaction && !getOptions().getAutoStartTransaction();
+    }
 
     private void cleanupTransaction() {
         inTransaction = false;
         transactionOptions = null;
-        if (getOptions().getAutoStartTransaction()) {
-            startTransaction(getOptions().getDefaultTransactionOptions());
-        }
         getServerSession().advanceTransactionNumber();
     }
-
-    private ReadPreference getTransactionReadPreferenceOrPrimary() {
```

Comment:
Inlined this method (in a separate commit).  It used to serve a purpose when we were supporting transactions on secondaries, but now it makes no sense.